Close #11614: Require plugins to specify their license (#11951)
OpenRCT2/OpenRCT2: distribution/scripting.md

Remote scripts on the other hand can alter the game state in certain contexts, thus must be enabled for every player in a multiplayer game. Players **cannot** enable or disable remote scripts for multiplayer servers they join. Instead the server will upload any remote scripts that have been enabled on the server to each player. This allows servers to enable scripts without players needing to manually download or enable the same script on their end.

> The authors should also define a valid license for the plugin, making it clear to the community whether that plugin can be altered, copied, etc. OpenRCT2 supports all licenses listed on [choosealicense](https://choosealicense.com/appendix/), make sure you pick one of them and use its identifier, as listed on [SPDX](https://spdx.org/licenses/).

Reviewer: OpenRCT2 does not "support" any licence.

If we are to _suggest_ a licence to people who don't know yet, make it MIT, I say - it's very common and allows others do as they please, including linking to GPL-incompatible software.

Author: I changed it to `MIT` in he example and rewrote this section slightly.

> The authors should also define a licence for the plugin, making it clear to the community whether that plugin can be altered, copied, etc. A good reference material is listed on [ChooseALlicense](https://choosealicense.com/appendix/), try to pick one of them and use its corresponding identifier, as listed on [SPDX](https://spdx.org/licenses/).

Reviewer: ```suggestion
The authors must also define a licence for the plug-in, making it clear to the community whether that plug-in can be altered, copied, etc. A good reference material is listed on [ChooseALlicense](https://choosealicense.com/appendix/), try to pick one of them and use its corresponding identifier, as listed on [SPDX](https://spdx.org/licenses/).
```

    version: '1.0',
    authors: ['Your Name'],
    type: 'remote',

>     license: 'MIT',

Reviewer 2: The code is using `licence`, but the example uses `license`. We should probably support both, but:
```suggestion
    licence: 'MIT',
```

>     license: 'Unlicense',

Reviewer 2: ```suggestion
    license: 'Unlicensed',
```
?

Author: I was really going for the [Unlicense](https://choosealicense.com/licenses/unlicense/), but I changed the example to `MIT` as per Michael's suggestion

Reviewer 2: Ah, I wasn't familiar with that one. I think we shouldn't specify any licence by default, though. Instead, force the plugin author to specify one — and have the TypeScript transpiler throw an error if a non-empty string is provided, or some such.

Reviewer: No. This refers to "The Unlicense".

Reviewer: Personally, I think providing a little bit of guidance would be OK. Perhaps we should:
- throw an error if the field is missing or empty
- have it empty by default, so that everyone *has* to pick one
- suggest MIT in the .md file if the user is unsure

What do you think?
